src/main/java/bhs/devilbotz/commands/auto/DriveStraightToDock.java

```diff
@@ -4,95 +4,103 @@
 
 package bhs.devilbotz.commands.auto;
 
-import bhs.devilbotz.Robot;
 import bhs.devilbotz.subsystems.DriveTrain;
-import edu.wpi.first.math.controller.PIDController;
-import edu.wpi.first.math.filter.SlewRateLimiter;
-import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
-import edu.wpi.first.wpilibj2.command.CommandBase;
 
 /**
  * This command is a PID controller that drives the robot straight to a set distance up a hill to
  * dock with the charging port.
  */
-public class DriveStraightToDock extends CommandBase {
-  private DriveTrain drivetrain;
-  private PIDController distancePid;
-  private PIDController straightPid;
-  private double distance;
-  private double startAngle;
-  private final SlewRateLimiter speedSlewRateLimiter = new SlewRateLimiter(1);
+public class DriveStraightToDock extends DriveStraightPID {
+  private enum DockState {
+    ON_GROUND,
+    ON_RAMP,
+    LEVELING_OFF
```

Comment:
We try to detect the various states we are in....approaching dock...on ramp....levelling off.